### DISCUSSION TITLE
How to bind <ESC> to exit autocomplete and enter vi mode?

### DISCUSSION THREAD

**Discussion by vulns-are-features-too** [marlonrichert/zsh-autocomplete#158] [Q&A] [answered]
I want `<Esc>` to exit autocomplete *and* enter vi mode. How should I bind my `<Esc>` key?

**Comment by marlonrichert** [comment, marked as answer]
If by "exit autocomplete" you mean "exit the completion menu" and by "vi mode" you mean `vi-cmd-mode`, then the following should do the trick:
```zsh
bindkey -M menuselect '^[' .vi-cmd-mode
```
